Sources/AWSLambdaRuntimeCore/HTTPClient.swift

```diff
@@ -43,13 +43,16 @@ internal final class HTTPClient {
                              timeout: timeout ?? self.configuration.requestTimeout))
     }
 
-    func post(url: String, body: ByteBuffer?, timeout: TimeAmount? = nil) -> EventLoopFuture<Response> {
-        self.execute(Request(targetHost: self.targetHost,
-                             url: url,
-                             method: .POST,
-                             headers: HTTPClient.headers,
-                             body: body,
-                             timeout: timeout ?? self.configuration.requestTimeout))
+    func post(url: String, body: ByteBuffer?, timeout: TimeAmount? = nil, additionalHeaders: HTTPHeaders? = nil) -> EventLoopFuture<Response> {
+        var headers = HTTPClient.headers
+        additionalHeaders.flatMap { headers.add(contentsOf: $0) }
```

Comment:
@tomerd I would vote to inject all headers on every call and to not have them split over two files. wdyt? In hindsight having static headers directly on the HTTPClient doesn't make much sense IMHO.

Reply:
hi @fabianfett do you mean have `RuntimeClient` fully control the headers instead of injecting "additional" ones? id so, that sounds good to me.

Reply:
Would it be ok to include that change in this PR or should I make a separate one for this?

Reply:
I suggest we fix it in this PR

Reply:
Ok. I moved the static header declaration into an extension on `RuntimeClient`. The `get` and `post` methods on `HTTPClient` now have a non-optional `headers` parameter (without a default value). I also moved the `headers` parameter to come after the `url`parameter ... that ordering made a more cohesive impression to me ^^